### PR TITLE
soc: st: stm32: stm32l4x: enable ART flash cache accelerator

### PR DESCRIPTION
**Pull request by weinholt**
Enable instruction cache, data cache and prefetching.

I am using an stm32l412 based board as an I²C target and noticed that there was quite a bit of clock stretching happening. Enabling the ART Accelerator improves the situation significantly, likely because the CPU is no longer stalling while fetching code/data from the flash.

I don't know why stm32l4x wasn't part of https://github.com/zephyrproject-rtos/zephyr/pull/36277. I looked at RM0394 and didn't see anything to indicate that ART Accelerator is unsupported on any of the STM32L4x models. The flash_stm32l4x.c driver flushes the data/instruction caches on erase, but this has been a no-op until now.

### CODE REVIEW

**Review by erwango — COMMENTED** [review]
At first sight (and after browsing history), I don't see a good reason why this was not extended to L4.

Approved. Thanks for the change.

Note that ~3% improvment can be seen on `tests/benchmarks/latency_measure`

Edit: I'm running a non-regression campaign on our bench. I'm blocking the PR until we get the tests
Edit 2: Once approved we can extend to other similar STM32SoCs on which it is also applicable (G4, WL, WB, ..)

**Review by erwango — APPROVED** [review]

**Comment by erwango** [comment]
Tests performed (twister -e logging -e cmsis_dsp -T tests -T samples -p disco_l475_iot1 ). No regression detected

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**
